File: 08 - Python Programming/rock-paper-scissor.py

import random
## welcome

welcome = """
██████╗  ██████╗  ██████╗██╗  ██╗    ██████╗  █████╗ ██████╗ ███████╗██████╗     ███████╗ ██████╗██╗███████╗███████╗ ██████╗ ██████╗ ███████╗
██╔══██╗██╔═══██╗██╔════╝██║ ██╔╝    ██╔══██╗██╔══██╗██╔══██╗██╔════╝██╔══██╗    ██╔════╝██╔════╝██║██╔════╝██╔════╝██╔═══██╗██╔══██╗██╔════╝
██████╔╝██║   ██║██║     █████╔╝     ██████╔╝███████║██████╔╝█████╗  ██████╔╝    ███████╗██║     ██║███████╗███████╗██║   ██║██████╔╝███████╗
██╔══██╗██║   ██║██║     ██╔═██╗     ██╔═══╝ ██╔══██║██╔═══╝ ██╔══╝  ██╔══██╗    ╚════██║██║     ██║╚════██║╚════██║██║   ██║██╔══██╗╚════██║
██║  ██║╚██████╔╝╚██████╗██║  ██╗    ██║     ██║  ██║██║     ███████╗██║  ██║    ███████║╚██████╗██║███████║███████║╚██████╔╝██║  ██║███████║
╚═╝  ╚═╝ ╚═════╝  ╚═════╝╚═╝  ╚═╝    ╚═╝     ╚═╝  ╚═╝╚═╝     ╚══════╝╚═╝  ╚═╝    ╚══════╝ ╚═════╝╚═╝╚══════╝╚══════╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝
"""

## rock
rock_pic = """
    _______
---'   ____)
      (_____)
      (_____)
      (____)
---.__(___)
"""

## paper
paper_pic = """
     _______
---'    ____)____
           ______)
          _______)
         _______)
---.__________)
"""

## scissors
scissor_pic = """
    _______
---'   ____)____
          ______)
       __________)
      (____)
---.__(___)
"""

pic = [rock_pic, paper_pic, scissor_pic]
print(welcome, "\nIt's a rock-paper-scissor game!\nWIN +2\nTIED +1\nLOSE 0")

hand = ["rock", "paper", "scissor"]
i = 0
user_total = 0
com_total = 0
score = [[1, 0, 2], [2, 1, 0], [0, 2, 1]]

con_stop = input("Do you want to continue or stop? ")
# loop
while con_stop != "stop":
  print(f"\nRound {i+1}")
  # user hand
  user_input = str(input("What do you choose? "))
  user_index = hand.index(user_input)
  user_hand = hand[user_index]

  # computer hand
  com_index = random.randint(0, 2)
  com_hand = hand[com_index]

  print(f"Your hand: {pic[user_index]}")
  print(f"Computer'hand: {pic[com_index]}")

  user_score = score[user_index][com_index]
  print(f"You got + {user_score}")
  user_total += user_score
  
  com_score = score[com_index][user_index]
  com_total += com_score

  con_stop = input("Do you want to continue or stop? ")
  i += 1

print(f"\nYour total score: {user_total}")
print(f"Computer's total score: {com_total}")
if user_total > com_total:
  print("You win! Congratulation ✨")
elif user_total == com_total:
  print("It's a tied match 🥊")
else:
  print("You lost. Wanna give another try? 🔥")

File: 03 - R Programming/rock-paper-scissor.r

game <- function(){

  flush.console()
  cat("It's a Rock-Paper-Scissor Game ✂️", "\nWIN +2\nTIED +1\nLOSE 0\n")

  flush.console()
  con <- readline("Do you want to continue or stop?: ")

  hand_to_num <- function(hand){
      if(hand=="rock"){
        return(1)
      }else if (hand=="paper"){
        return(2)
      }else{
        return(3)
      }
  }

  score_matrix <- matrix(c(1,2,0,0,1,2,2,0,1), nrow = 3, ncol = 3)

  all_user_score <<- 0
  all_com_score <<- 0
  i <- 1

  while (con != "stop"){

  # USER PART
  flush.console()
  cat("\nRound ", i, "\n")
  flush.console()
  user_text <- tolower(as.character(readline("Your hand?: ")))
  user_hand <- hand_to_num(user_text)

  flush.console()
  #COM PART
  all_hands <- c("rock", "paper", "scissor")
  com_text <- sample(all_hands, 1)
  cat("Computer's hand: ", com_text, "\n")
  com_hand <- hand_to_num(com_text)

  flush.console()
  #SCORE
  user_score <- score_matrix[user_hand, com_hand]
  cat("You got ", user_score, "\n")
  all_user_score <<- user_score + all_user_score
  com_score <- score_matrix[com_hand, user_hand]
  all_com_score <<- com_score + all_com_score

  flush.console()
  # PLAY AGAIN OR NOT
  con <- readline("Do you want to continue or stop?: ")
  i <- i+1
  }

  flush.console()
  cat("\nYour total score: " , all_user_score, "\nComputer's total score: ", all_com_score)
  if (all_user_score > all_com_score){
    cat("\nYou win! Congratulation ✨")
  }else if (all_user_score < all_com_score){
    cat("\nYou lost. Wanna give another try? 🔥")
  }else{
    cat("\nIt's a tied match 🥊")
  }

}


# CALL FUNCTION
game()
